### PR TITLE
feat: If a PR has oss.gg and points assigned as labels, it should auto award on merge

### LINKED ISSUE #86: [FEATURE] If a PR has oss.gg and points assigned as labels, it should auto award on merge
### Is your feature request related to a problem? Please describe.

currently, we have to award manually (and sometimes we forget 😇 )

### Describe the solution you'd like

if a PR has oss.gg label and points label lets auto award points and post a comment on the PR so everyone is informed :)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by RajuGangitla**
## What does this PR do?

If a PR has oss.gg and points assigned as labels, it should auto award on merge

Fixes #86 

https://github.com/user-attachments/assets/f3b692b0-6f64-4845-a436-313d2c99eb1b

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at oss.gg](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: lib/github/hooks/issue.ts [hunk 1]
```diff
@@ -427,6 +426,99 @@ export const onPullRequestOpened = async (webhooks: Webhooks) => {
...
+    const owner = context.payload.repository.owner.login;
+
+    // Early exit if the pull request was not merged
+    if (!pullRequest.merged) {
+      console.log("Pull request was not merged.");
+      return;
+    }
+
+    const octokit = getOctokitInstance(context.payload.installation?.id!);
+
```
Comment (jobenjada) [inline lib/github/hooks/issue.ts:442]: let's first check if the PR itself has a points label and if so, award the points. If not, go and search in linked issues.

Comment (jobenjada) [reply lib/github/hooks/issue.ts:442]: make sure to not repeat the code for awarding points (and if the user isn't found creating one).

we already have that functionality in the /award function. You might have to move it to the utils.ts in the /github folder and then refactor both the onAwardPoints and your new function

File: lib/github/hooks/issue.ts [hunk 2]
```diff
@@ -427,6 +426,99 @@ export const onPullRequestOpened = async (webhooks: Webhooks) => {
...
+        const issueLabels = issue.labels.map((label: { name: string }) => label.name);
+
+        // Check if the "🕹️ oss.gg" label is present, skip if not
+        const ossLabel = issueLabels.find((label) => label.includes(OSS_GG_LABEL));
+        if (!ossLabel) {
+          console.log(`Issue #${issueNumber} does not have the 🕹️ oss.gg label. Skipping.`);
+          continue;  // Skip this issue and move to the next one
+        }
+
+        // Extract points from labels like "🕹️ 50 points", "🕹️ 100 points", etc.
```
Comment (jobenjada) [inline lib/github/hooks/issue.ts:477]: we have a `extractPointsFromLabel` helper function in github/utils - please use this

File: lib/github/hooks/issue.ts [hunk 3]
```diff
@@ -427,6 +426,99 @@ export const onPullRequestOpened = async (webhooks: Webhooks) => {
...
+          // Fetch PR author details only if the oss.gg label is present
+          const prAuthorGithubId = pullRequest.user.id;
+          const prAuthorUsername = pullRequest.user.login;
+          const { data: prAuthorProfile } = await octokit.users.getByUsername({
+            username: prAuthorUsername,
+          });
+
+          // Fetch or create user profile only if the oss.gg label is present
+          let user = await getUserByGithubId(prAuthorGithubId);
+          if (!user) {
```
Comment (jobenjada) [inline lib/github/hooks/issue.ts:495]: as mentioned above, please create a helper function and refactor onAwardPoints to use it

File: lib/github/hooks/issue.ts [hunk 4]
```diff
@@ -427,6 +426,99 @@ export const onPullRequestOpened = async (webhooks: Webhooks) => {
...
+          }
+
+          // Award points to the user
+          await assignUserPoints(
+            user?.id,
+            points,
+            "Awarded points",
+            pullRequest.html_url,
+            ossGgRepo?.id
+          );
```
Comment (jobenjada) [inline lib/github/hooks/issue.ts]: after the points are awarded, we want to add the same comment to the PR we issue on onAwardPoints. Here, also make sure to follow the DRY coding principle.

**Review by jobenjada — CHANGES_REQUESTED** [review]

**Comment by RajuGangitla** [comment]
@jobenjada pushed the latest changes

**Comment by jobenjada** [comment]
code looks good, will test and merge likely on the weekend as I'm not sure if I have time before hand 🙌

**Review by jobenjada — APPROVED** [review]
LGTM

**Pull request merged**